File: app/models/question_visit.rb
class QuestionVisit < Prequel::Record
  column :id, :integer
  column :question_id, :integer
  column :user_id, :integer
  column :created_at, :datetime
  column :updated_at, :datetime

  belongs_to :question
  belongs_to :user

  def team_ids
    question ? question.team_ids : []
  end
end

File: app/assets/javascripts/models/question_visit.js
//  Copyright (c) 2010-2011, Nathan Sobo and Max Brunsfeld.  This file is
//  licensed under the Affero General Public License version 3 or later.  See
//  the COPYRIGHT file.

QuestionVisit = Monarch("QuestionVisit", {
  userId: 'key',
  questionId: 'key',
  updatedAt: 'datetime'
})
  .belongsTo('question')
  .belongsTo('user');

